internal/provider/resource_vcf_instance.go

```diff
@@ -0,0 +1,349 @@
+	}
+	if excludedComponentsAttribute, ok := d.GetOk("excluded_components"); ok {
+		sddcSpec.ExcludedComponents = utils.ToStringSlice(excludedComponentsAttribute.([]interface{}))
+	}
+
+	bringUpInfo, err := getLastBringUp(ctx, client)
+	if err != nil {
+		tflog.Error(ctx, err.Error())
+		return diag.FromErr(err)
+	}
+
+	bringUpID, diags := invokeBringupWorkflow(ctx, client, sddcSpec, bringUpInfo)
+	if diags != nil {
+		return diags
+	}
+
+	return waitForBringupProcess(ctx, bringUpID, client)
+}
+func resourceVcfInstanceRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*api_client.CloudBuilderClient)
+
+	bringUpInfo, err := getLastBringUp(ctx, client)
+	if err != nil {
+		tflog.Error(ctx, err.Error())
+		return diag.FromErr(err)
+	}
+
+	d.SetId(bringUpInfo.ID)
+	_ = d.Set("status", bringUpInfo.Status)
+	_ = d.Set("creation_timestamp", bringUpInfo.CreationTimestamp)
+	return nil
+}
+func resourceVcfInstanceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// no op
+	return resourceVcfInstanceRead(ctx, d, meta)
+}
+func resourceVcfInstanceDelete(_ context.Context, _ *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	// no op
+	return nil
+}
+
+func invokeBringupWorkflow(ctx context.Context, client *api_client.CloudBuilderClient, sddcSpec *models.SDDCSpec, lastBringup *models.SDDCTask) (string, diag.Diagnostics) {
+	var bringUpID string
+	if lastBringup != nil && lastBringup.Status != "COMPLETED_WITH_SUCCESS" {
+		bringUpID = lastBringup.ID
+		diags := validateBringupSpec(ctx, client, sddcSpec, true)
+		if diags != nil {
+			return bringUpID, diags
+		}
+
+		retryBringupParams := sddc_api.NewRetrySDDCParamsWithContext(ctx).
+			WithTimeout(constants.DefaultVcfApiCallTimeout).WithID(bringUpID).WithSDDCSpec(sddcSpec)
+		okResponse, acceptedResponse, err := client.ApiClient.SDDC.RetrySDDC(retryBringupParams)
+		if okResponse != nil {
+			bringUpID = okResponse.Payload.ID
+		}
+		if acceptedResponse != nil {
+			bringUpID = acceptedResponse.Payload.ID
+		}
+		if err != nil {
+			return "", diag.FromErr(err)
+		}
+	} else {
+		diags := validateBringupSpec(ctx, client, sddcSpec, false)
+		if diags != nil {
+			return bringUpID, diags
+		}
+
+		bringupParams := sddc_api.NewCreateSDDCParamsWithContext(ctx).
+			WithTimeout(constants.DefaultVcfApiCallTimeout).WithSDDCSpec(sddcSpec)
+
+		okResponse, acceptedResponse, err := client.ApiClient.SDDC.CreateSDDC(bringupParams)
+		if okResponse != nil {
+			bringUpID = okResponse.Payload.ID
+		}
+		if acceptedResponse != nil {
+			bringUpID = acceptedResponse.Payload.ID
+		}
+		if err != nil {
+			return "", diag.FromErr(err)
+		}
+
+		tflog.Info(ctx, fmt.Sprintf("Bring-Up workflow with ID %s has started", bringUpID))
+	}
+	return bringUpID, nil
+}
+
+func waitForBringupProcess(ctx context.Context, bringUpID string, client *api_client.CloudBuilderClient) diag.Diagnostics {
```

Comment:
Shall we put a timeout?

Reply:
Good question. Perhaps a default timeout but allow an override?

Reply:
There is a 4 hour timeout defined at Terraform SDK level. See Timeouts in ResourceVcfInstance()